### PR TITLE
Refactored Dolibarr triggers

### PR DESCRIPTION
**Pull request by marcosgdf**
Refactored Dolibarr triggers:
Created a DolibarrTriggers abstract class.
Moved common variables out of construct class
Improved run_trigger method signature
Versions are now constants of DolibarrTriggers class.

### CODE REVIEW

**Comment by marcosgdf** [comment]
The rest of CS errors are because run_trigger is not camelCased, but that's not a choice right now...

File: htdocs/core/triggers/DolibarrTriggers.class.php [hunk 1]
```diff
@@ -0,0 +1,149 @@
+<?php
```
Comment (rdoursenaud) [inline htdocs/core/triggers/DolibarrTriggers.class.php:1]: I would have named this class just Trigger, we know we are in Dolibarr ;)
Also, I hope we'll add the Dolibarr namespace someday!

**Comment by rdoursenaud** [comment]
Great work BTW!

File: htdocs/core/triggers/DolibarrTriggers.class.php (hunk 1 above)
Comment (marcosgdf) [inline htdocs/core/triggers/DolibarrTriggers.class.php:1]: Well I called DolibarrTriggers inspired by the DolibarrModules class name

Comment (rdoursenaud) [inline htdocs/core/triggers/DolibarrTriggers.class.php:1]: Then we'll refactor both when we'll add namespaces :p

**Pull request merged**